### PR TITLE
Add ability to set cursor position

### PR DESCRIPTION
**Pull request by ezraberch**
This is a companion PR to https://github.com/CodeEditApp/CodeEdit/pull/840, adding the necessary functionality to implement https://github.com/CodeEditApp/CodeEdit/issues/788.

### CODE REVIEW

**Review by thecoolwinter — CHANGES_REQUESTED** [review]
This change looks like a good idea. I have limited experience with Combine, so I'm assuming the cursor updating functions correctly. I've requested a couple small changes. First is you need to make the `cursorPosition` init parameter optional and with a default value so we don't break CodeEdit (or any other project depending on CodeEditTextView) by adding a variable that it doesn't handle yet. There's also just a couple lines that are >120 characters that I'd like shortened before I can approve the PR.

Looks like a good change! Thanks for the work you've put into it!

**Review by lukepistrol — CHANGES_REQUESTED** [review]
There are multiple SwiftLint errors. Please run SwiftLint on your machine and make sure to resolve all the errors.

Screenshot 2022-10-28 at 11 56 12

File: Sources/CodeEditTextView/STTextViewController.swift [hunk 1]
```diff
@@ -223,6 +228,35 @@ public class STTextViewController: NSViewController, STTextViewDelegate, ThemeAt
...
+
+        var (line, column) = position
+        let string = textView.string
+        if (line > 0) {
+            string.enumerateSubstrings(in: string.startIndex..<string.endIndex) { substring, substringRange, _, done in
+                line -= 1
+                if line < 1 {
+                    // If `column` exceeds the line length, set cursor to the end of the line.
+                    let index = min(substringRange.upperBound, string.index(substringRange.lowerBound, offsetBy: column - 1))
+                    if let newRange = NSTextRange(NSRange(index..<index, in: string), provider: provider) {
```
[the remark this answers is not in the document]
Comment (lukepistrol) [reply Sources/CodeEditTextView/STTextViewController.swift:249]: We actually run the linter as part of our CI

**Review by lukepistrol — APPROVED** [review]
LGTM 👍

**Review by lukepistrol — CHANGES_REQUESTED** [review]

**Review by lukepistrol — APPROVED** [review]

**Review by thecoolwinter — APPROVED** [review]
LGTM!

**Pull request merged**